utils/misc/misc.go

```diff
@@ -296,12 +298,35 @@ func RemoveFilePaths(filePaths ...string) {
 	}
 }
 
+// GetTempFolders returns all temporary folders.
+func GetTempFolders() (tempFolders []string) {
+	tmpDirPath, err := CreateTMPDIR()
+	if err != nil {
+		return
+	}
+
+	tempFolders = []string{
+		tmpDirPath,
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-async-destination-logs"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-archives"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-warehouse-staging-uploads"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-raw-data-destination-logs"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-warehouse-load-uploads-tmp"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-identity-merge-rules-tmp"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-identity-mappings-tmp"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-redshift-manifests"),
+		fmt.Sprintf("%s/%s", tmpDirPath, "rudder-warehouse-json-uploads-tmp"),
+		fmt.Sprintf("%s/%s", tmpDirPath, config.GetEnv("RUDDER_CONNECTION_TESTING_BUCKET_FOLDER_NAME", "rudder-test-payload")),
+	}
+	return
+}
+
 // RemoveEmptyFolderStructureForFilePath recursively cleans up everything till it reaches the stage where the folders are not empty or parent.
 func RemoveEmptyFolderStructureForFilePath(fp string) {
 	if fp == "" {
 		return
 	}
-	for currDir := filepath.Dir(fp); currDir != "/" && currDir != "."; {
+	for currDir := filepath.Dir(fp); currDir != "/" && currDir != "." && !Contains(tempFolders, currDir); {
```

Comment:
@achettyiitr just for my understanding, what was the issue with current code. `rmDir` would only delete empty folders, right?

Reply:
Actually the issue was, When we were in the deletion flow we were stopping at the root.
Let's say that the time window start at t1 and completes at t2. Before t1 if somebody has created the folders, so we would have deleted it during this time interval. Now after t2, if we are trying to access the files we will get error.